From 3d3078e10352d845d26c108ebcdf62f3bff0c3fe Mon Sep 17 00:00:00 2001
From: Adrian Freihofer <adrian.freihofer@siemens.com>
Date: Tue, 28 Dec 2021 21:11:59 +0000
Subject: [PATCH 4/4] configure: libpci detection optional

For native builds where only tools are built libpci detection must be
disabled.

Signed-off-by: Adrian Freihofer <adrian.freihofer@siemens.com>
---
 configure.ac | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index 01a4fdb..da15264 100644
--- a/configure.ac
+++ b/configure.ac
@@ -179,7 +179,14 @@ if test "x$PKG_CONFIG" = "xno"; then
 	AC_MSG_ERROR([You need to install pkg-config])
 fi
 PKG_CHECK_MODULES(LIBCHECK, check)
-PKG_CHECK_MODULES(LIBPCI, libpci)
+
+AC_ARG_ENABLE([libpci],
+    AS_HELP_STRING([--disable-libpci], [Disable libpci detection if only tools need to be built]))
+
+AS_IF([test "x$enable_libpci" != "xno"], [
+    PKG_CHECK_MODULES(LIBPCI, libpci)
+])
+
 # ------------------------------------------------------------------------------
 AC_CONFIG_FILES([
 	Makefile
-- 
2.31.1

